docs/LabVIEW_Debugging_Techniques.md

```diff
@@ -0,0 +1,95 @@
+Once added to the system definition, custom devices have been fully integrated into VeriStand’s context. As a result, LabVIEW’s built-in Debugging Techniques are no longer available. Several techniques are available for debugging and benchmarking the custom device.
+ 
+#### Console Viewer
+
+A subcomponent of VeriStand RT Engine is the RT Console Viewer.
+You can install it to the execution host using Measurement and Automation Explorer. When installed, the component runs a small UDP daemon allowing the operator to view the console from a utility called the RT Console Viewer. You can access the RT Console viewer from the **VeriStand Editor » Tool Launcher » View Console**.
+
+![](images/ConsoleViewer1.jpg)
+
+**Figure: RT Console Viewer**
+
+The Console Viewer will show the system definition and the resulting CPU usage. The viewer is also useful for displaying debugging messages. The console viewer provides a periodic snapshot of utilization. CPU spikes and transients will probably be unobservable. If the system is very busy, it may not update the console viewer at all. You can use other debugging methods for a more accurate indication of resource utilization.
+
+As the name implies, the RT Console Viewer is only available on real-time targets. The RT Console Viewer is also available as a stand-alone add-on to LabVIEW Real-Time. See **[NI Developer Zone Tutorial: Remotely View Console Output of Real-Time Targets](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA03q000000x4TjCAI&l=en-US)** for more information.
+
+&nbsp;&nbsp;&nbsp; **Note:** You cannot use the **Console Viewer** on NI Linux Real-Time targets. Instead, connect your NI Linux Real-Time targets to a computer using a serial port to view the output.
+
+#### Printing to the Console
+Printing to the console is often all that’s needed to debug an application.
+
+#### Printing With "NI VeriStand - Print Debug Line" VI
+The recommended method of printing to the console is to use "NI VeriStand - Print Debug Line" VI. This VI is part of Custom Device API » Utilities » Print Debug Line.vi.
+
+![](images/NIVSdebugString.jpg)
+
+This VI works on both Windows and RT execution hosts. It has an optional input (Attribute) to change the color of the text. It also has an optional input to append the string to the VeriStand log file.
+
+#### Printing With ni_emb.dll
+
+You’ll find **ni_emb.dll** in **&lt;labview&gt;\Targets\NI\RT\vi.lib**. This dll contains a stub function called **PrintStringToConsole**. Calling this function sends a string to the RT console. Configure the function to run in any thread using the C calling convention. The return type is void and it has a C String pointer input constant. You’ll find this function wrapped in a VI in the same folder in **rtutility.llb\RT Debug String.vi**. Since **ni_emb.dll** is a stub dll, it’s not necessary to deploy this VI to the RT target. The stub exists so the **PrintStringToConsole** function does not return an error when called on Windows.
```

Comment:
we should not be talking about call library nodes that call .dlls.  those won't work on linux, and it looks like there's no ni_emb equivalent on linux anyway.  it is an outdated implementation detail that has been buried under another reentrant VI anyway (NI_SystemLogging.lvlib:Write to System Log.vi)

we should instead advise calling the RT Debug String.vi  and update its location to be correct, which is at: 
C:\Program Files (x86)\National Instruments\LabVIEW 2020\Targets\NI\RT\vi.lib\rtutility.llb\RT Debug String.vi

this VI will work on all RT platforms and means we don't have to talk about conditional disable structures at all.